cmd/govim/internal/util/util.go

```diff
@@ -0,0 +1,65 @@
+package util
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// Split breaks the line into words, evaluating quoted
+// strings and evaluating environment variables.
+// Implementation is a slightly modified version of go generate's
+// split(), from src/cmd/go/internal/generate/generate.go in stdlib
+func Split(line string) ([]string, error) {
+	// Parse line, obeying quoted strings.
+	var words []string
+	// There may still be a carriage return.
+	if len(line) > 0 && line[len(line)-1] == '\r' {
+		line = line[:len(line)-1]
+	}
+	// One (possibly quoted) word per iteration.
+Words:
+	for {
+		line = strings.TrimLeft(line, " \t")
+		if len(line) == 0 {
+			break
+		}
+		if line[0] == '"' {
+			for i := 1; i < len(line); i++ {
+				c := line[i] // Only looking for ASCII so this is OK.
+				switch c {
+				case '\\':
+					if i+1 == len(line) {
+						return words, fmt.Errorf("bad backslash")
+					}
+					i++ // Absorb next byte (If it's a multibyte we'll get an error in Unquote).
+				case '"':
+					word, err := strconv.Unquote(line[0 : i+1])
+					if err != nil {
+						return words, fmt.Errorf("bad quoted string")
+					}
+					words = append(words, word)
+					line = line[i+1:]
+					// Check the next character is space or end of line.
+					if len(line) > 0 && line[0] != ' ' && line[0] != '\t' {
+						return words, fmt.Errorf("expect space after quoted argument")
+					}
+					continue Words
+				}
+			}
+			return words, fmt.Errorf("mismatched quoted string")
+		}
+		i := strings.IndexAny(line, " \t")
+		if i < 0 {
+			i = len(line)
+		}
+		words = append(words, line[0:i])
+		line = line[i:]
+	}
+	// Substitute environment variables.
+	for i, word := range words {
```

Comment:
We don't want to expand in this case, because values will already have been expanded in the setting of `GOVIM_GOPLS_FLAGS`. Expansion is required in `go generate` because we are evaluating literals in a `.go` file. So this loop can be dropped

Reply:
fixed